src/cartesian/Area.tsx

```diff
@@ -554,20 +554,35 @@ export class Area extends PureComponent<Props, State> {
     const { isAnimationFinished } = this.state;
     const hasSinglePoint = points.length === 1;
     const layerClass = classNames('recharts-area', className);
-    const needClip = (xAxis && xAxis.allowDataOverflow) || (yAxis && yAxis.allowDataOverflow);
+    const needClipX = xAxis && xAxis.allowDataOverflow;
+    const needClipY = yAxis && yAxis.allowDataOverflow;
+    const needClip = needClipX || needClipY;
     const clipPathId = _.isNil(id) ? this.id : id;
+    const { r, strokeWidth } = filterProps(dot) || { r: 3, strokeWidth: 2 };
+    const { clipDot = true } = dot as DotProps;
```

Comment:
any way we can avoid casting this? Can we create a typeguard?

add to `DotProps` a 
```
...
propsType: 'dot';
...
```

then

`const isDotProps = (dot: AreaDot): dot is DotProps => 'propsType' in dot && dot.propsType === 'dot'`?